src/sagas/tokens.js

```diff
@@ -281,6 +283,48 @@ export function* monitorSelectedToken() {
   }
 }
 
+/**
+ * This saga will create a channel to queue PROPOSAL_TOKEN_FETCH_REQUESTED actions and
+ * consumers that will run in parallel consuming those actions.
+ *
+ * More information about channels can be read on https://redux-saga.js.org/docs/api/#takechannel
+ */
+function* fetchProposalTokenDataQueue() {
+  const fetchProposalTokenDataChannel = yield call(channel);
+
+  // Fork CONCURRENT_FETCH_REQUESTS threads to download token balances
+  for (let i = 0; i < METADATA_CONCURRENT_DOWNLOAD; i += 1) {
```

Comment:
I think we should use a smaller quantity here. The metadata is requested from the explorer service, the token data is requested from the full node. Why don't we start with 3 or 5? It should be hard to have any tx with more than this amount of different tokens.

Reply:
Good point, on [a07b2cc](https://github.com/HathorNetwork/hathor-wallet/pull/401/commits/a07b2cc965d57036cdad292ed80b63884c8f31a5) changed it to the `CONCURRENT_FETCH_REQUESTS` constant already used for fetching token metadata from the fullnode. Currently its value is `5`.